File: NextGreaterElement.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = [-1] * len(nums1) 
        d = {val: idx for idx, val in enumerate(nums1)}

        for i in range(len(nums2)):
            n = nums2[i]
            if n in d:
                for j in range(i+1, len(nums2)):
                    if nums2[j] > n:
                        res[d[n]] = nums2[j]
                        break

        return res

File: NextGreaterElement.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> dict = new HashMap<>();
        Stack<Integer> stack = new Stack<>();
        int manupilativeIdx;
        int[] res = new int[nums1.length];
        Arrays.fill(res, -1);
        for (int i = 0; i < nums1.length; i++){
            dict.put(nums1[i], i);
        }
        for (int i = 0; i < nums2.length; i++){
          while (!stack.isEmpty() && nums2[i] > stack.peek()){
              manupilativeIdx = dict.get(stack.pop());
              res[manupilativeIdx] = nums2[i];
          }
            if (dict.containsKey(nums2[i])){
                stack.push(nums2[i]);
            }

        }
        return res;
    }
}
